frontend/src/components/ecommerce/PayWithVipps.tsx

```diff
@@ -0,0 +1,51 @@
+import { ApolloError, useMutation } from "@apollo/client";
+import { INITIATE_ORDER } from "@graphql/ecommerce/mutations";
+import { Card, CardActionArea, CardMedia, CircularProgress, makeStyles, Theme } from "@material-ui/core";
+import { useRouter } from "next/router";
+import React from "react";
+
+const useStyles = makeStyles((_: Theme) => ({
+  root: {
+    maxWidth: 300,
+    background: "inherit",
+  },
+}));
+
+interface Props {
+  productId: string;
+  quantity: number;
+  onError?: (e?: ApolloError) => void;
+}
+
+const PayWithVipps: React.FC<Props> = ({ productId, quantity, onError }) => {
+  const [initiateOrder, { loading, error }] = useMutation(INITIATE_ORDER, {
+    onCompleted: (data) =>
+      router.push(data.initiateOrder.redirect || `/ecommerce/fallback?orderId=${data.initiateOrder.orderId}`),
+    onError: onError,
+  });
+
+  const classes = useStyles();
+  const router = useRouter();
+
+  if (loading) return <CircularProgress />;
```

Comment:
If possible, removing this will allow Next to statically build the components, improving performance